App/Product/frmProduct.cs

```diff
@@ -26,6 +27,7 @@ private void btnEdit_Click(object sender, EventArgs e)
         {
             if (selectedProductId != 0)
             {
+                if (MessageBox.Show("Do you want to edit the selected product?", "Edit Product", MessageBoxButtons.YesNo, MessageBoxIcon.Question) == DialogResult.Yes)
```

Comment:
This code block is missing an opening and ending curly brace. This will result in executing the rest of the code regardless of the actions (i.e. YES or NO).